### PR TITLE
feat: make abusing eslint-disable more difficult

### LINKED ISSUE #204: abusing eslint-disable should be more difficult to accomplish
We currently have a high rate of inlined `eslint-disable`s that are used to circumvent the linter when fixing the underlying issue is preferable.

We currently use `eslint-plugin-unicorn`'s `no-abusive-disable` rule to prevent users from using `/* eslint-disable */`, but this doesn't prevent users from disabling rules for an entire file rather than disabling them for a problematic section, nor does it incentivize upstreaming changes to rules that don't work reliably.

We should investigate/use [`eslint-plugin-eslint-comments`](https://mysticatea.github.io/eslint-plugin-eslint-comments/), which will prevent a large class of `eslint-disable` errors and generally improve code quality and our shared linter configuration.

### PR DESCRIPTION
**Pull request by ndhoule**
This changeset adds `eslint-plugin-eslint-comments` and a [set of rules](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/d2d21f2496232248370da851d7dbd634e2a37d87/lib/configs/recommended.js) intended to help prevent abuse of `eslint-disable` directives and to provide feedback when `eslint-disable` directives can be removed.

Closes #204.

### CODE REVIEW

File: src/config/recommended.js [hunk 1]
```diff
@@ -66,7 +72,37 @@ export default {
     'operator-assignment': ['error', 'always'],
     'spaced-comment': ['error', 'always'],
 
-    // eslint-plugin-import
+    /*
+     * eslint-plugin-eslint-comments
+     */
+    // Allow `eslint-disable`s at the top of a file, but prevent users from forgetting to close an
+    // `eslint-disable` in the middle of a file. We should consider whether or not we actually want
+    // to allow whole-file `eslint-disable`s, but in practice we do it frequently for somtimes-
```
Comment (serhalp) [inline src/config/recommended.js]: typo: `somtimes`

File: src/config/recommended.js [hunk 2]
```diff
@@ -66,7 +72,37 @@ export default {
...
+    // legitimate reasons.
```
Comment (serhalp) [inline src/config/recommended.js]: Yeah, I'm torn here too. A good example is the camelcase rule, in code that interacts with HJ or RS. Even if you factor that code well and isolate it to just a few files, there's no configuration I've seen or can think of that is really... perfect. 🤷‍♂

File: src/config/recommended.js [hunk 3]
```diff
@@ -66,7 +72,37 @@ export default {
...
+    'eslint-comments/no-use': [
+      'error',
+      {
+        allow: [
+          'eslint-disable',
+          'eslint-disable-line',
+          'eslint-disable-next-line',
+          'eslint-enable',
+          // Eventually, we should force users to specify all globals in their project's ESLint
+          // configuration file(s).
```
Comment (serhalp) [inline src/config/recommended.js]: // Eventually, we should force users to ~~specify all globals in their project's ESLint
// configuration file(s).~~
// never use globals. 😈

File: src/config/recommended.js [hunk 4]
```diff
@@ -102,12 +140,13 @@ export default {
     'lodash/preferred-alias': 'error',
     'lodash/prop-shorthand': ['error', 'never'],
 
-    // eslint-plugin-unicorn
+    /*
+     * eslint-plugin-unicorn
```
Comment (serhalp) [inline src/config/recommended.js]: We've got a lot of plugins in here now. They all document their rules in different ways and in different places. Consider adding a link to the rules documentation for each plugin in its comment heading!

Comment (ndhoule) [reply src/config/recommended.js]: There's a link to the plugin docs up in the `plugins` array.

Comment (ndhoule) [reply src/config/recommended.js]: (I'm cool moving documentation wherever makes sense, but it _is_ documented at least.)

**Comment by ndhoule** [comment]
We discussed on Slack, but for posterity: This actually represents our current best practices and this shouldn't require a ton of developer intervention—the only cases we'll really need to fix are where we've disabled all eslint rules for a file or region within a file, and when we've disabled eslint within a file but don't have a matching re-enable (a pretty straightforward, legitimate programmer error).

**Pull request merged**